src/main/java/com/google/cloud/tools/appengine/configuration/RunConfiguration.java

```diff
@@ -179,7 +180,7 @@ public String getPythonStartupArgs() {
     return pythonStartupArgs;
   }
 
-  @Nullable
+  /** @return a mutable list that exposes the internal state */
```

Comment:
I mean I the solution might be to make a copy of this class into the builder if you want to edit again. I understand that it was never immutable, but it feels weird to do this.

```
RunConfiguration.builder(runConfiguration)
```

wdyt?

Reply:
We can do that. It is a lot more work. Protobufs have this automatically. Here we have to write the code to copy everything. We also get hit because the builders don't have getter methods, though that's a little simpler to work around. There are some places we'll have to carry around both a builder and a run configuration.

Reply:
Why do the builders need getters? I'm a little confused, can you point me to the eclipse code?

Reply:
But I think using a getting on the builder might be the more meaningful solution if that's what you need

Reply:
https://github.com/GoogleCloudPlatform/google-cloud-eclipse/pull/3391 is where some of this work is happening. Look particularly at LocalAppEngineServerLaunchConfigurationDelegate and maybe LocalAppEngineServerBehavior. The main reason to have getters on builders ifs if we're passing around builders instead of RunConfigurations between methods, because we also need to query into values. Having all the setters in one class and the getters in another means we need to have two objects when we have code that needs to both read and write.